src/eth.rs

```diff
@@ -295,13 +295,9 @@ impl<'d> EthDriver<'d> {
     esp_idf_eth_spi_ethernet_ksz8851snl
 ))]
 impl<'d> EthDriver<'d> {
-    pub fn new_spi<P: spi::Spi + 'd>(
-        spi: impl Peripheral<P = P> + 'd,
+    pub fn new_spi(
+        spi_driver: &spi::SpiDriver<'d>,
```

Comment:
The SPI driver should be configured with DMA mode enabled and with transfer size > 1500. Shall we check that?

Reply:
Is it the case for all chips or just W5500?

Reply:
Pretty sure all chips.

Reply:
`max_transfer_size` is not exposed from `SpiDriver`. Any suggestions how I could assert it?

Reply:
Oh well. Forget it. Users will get a cryptic error message, they need to decypher. :) Exposing bits and pieces from the configuration of the `SpiDriver` just to assert it here does not feel OK.